driver/src/contracts/stablex_contract.rs

```diff
@@ -42,6 +45,59 @@ pub trait StableXContract {
     ) -> Result<()>;
 }
 
+struct AuctionElement {
+    valid_from: U256,
+    valid_until: U256,
+    sell_token_balance: u128,
+    order: Order,
+}
+
+impl AuctionElement {
+    fn in_auction(&self, index: U256) -> bool {
+        self.valid_from < index && index <= self.valid_until
+    }
+
+    fn from_bytes(bytes: &[u8; 113]) -> Self {
+        let mut data_vector = bytes.to_vec();
+        let account_id = H160::pop_from_log_data(&mut data_vector);
+        let sell_token_balance = u128::pop_from_log_data(&mut data_vector);
+        let buy_token = u16::pop_from_log_data(&mut data_vector);
+        let sell_token = u16::pop_from_log_data(&mut data_vector);
+        let valid_from = U256::from(u32::pop_from_log_data(&mut data_vector));
+        let valid_until = U256::from(u32::pop_from_log_data(&mut data_vector));
+        let is_sell_order = bool::pop_from_log_data(&mut data_vector);
+        let price_numerator = u128::pop_from_log_data(&mut data_vector);
+        let price_denominator = u128::pop_from_log_data(&mut data_vector);
+        let remaining_amount = u128::pop_from_log_data(&mut data_vector);
+
+        // Todo - Will likely have to compute this differently.
+        // TODO - put in own function fn [u128; 3] -> [u128; 2]
+        let buy_amount: u128;
+        let sell_amount: u128;
+        if is_sell_order {
+            sell_amount = remaining_amount;
+            buy_amount = (price_numerator * remaining_amount) / price_denominator;
+        } else {
+            buy_amount = remaining_amount;
+            sell_amount = (price_denominator * remaining_amount) / price_numerator;
+        }
+
+        AuctionElement {
+            valid_from,
+            valid_until,
+            sell_token_balance,
+            order: Order {
+                batch_information: None, // TODO - can't recover order_id from this information include it in the return of
```

Comment:
What would you suggest doing here? Should we add this to the data we return from the smart contract or count the number of items ourselves?

Reply:
Yes, this will require inclusion of more information being returned from the `getAuctionElement` method. Will need to update the contracts before we can finish this.

Reply:
I like the counting approach more...